### PR TITLE
homework 8

### PR DESCRIPTION
**Pull request by psergeev**

### CODE REVIEW

File: src/components/MovieCard/MovieCard.tsx [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
+                <span className={classes.title}>{movie.title}</span>
+                <span className={classes.date}>{movie.release_date.split('-')[0]}</span>
+            </div>
+            <div className={classes.genres}>{movie.genres.join(' & ')}</div>
+        </Card>
     );
-});
+}
+
+export default React.memo(withStyles(movieCardStyles)(MovieCard));
```
Comment (loginov-rocks) [inline src/components/MovieCard/MovieCard.tsx]: Prefer composition inside `index.ts`, for test you can still use `identity-obj-proxy` for `classes` property.

File: src/components/MovieCard/MovieCard.tsx [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
-import './MovieCard.scss';
+import movieCardStyles from './MovieCard.styles';
 
 interface Props extends RouteComponentProps<any> {
     handleMovieCardClick: (title: string) => void;
     movie: Movie;
 }
 
-export default React.memo(({ handleMovieCardClick, movie, history }: Props) => {
+type PropsWithStyles = Props & WithStyles<'card' | 'media' | 'titleDate' | 'title' | 'date' | 'genres'>;
```
Comment (loginov-rocks) [inline src/components/MovieCard/MovieCard.tsx:12]: Not recommend to type properties as string literals not to miss them during refactoring, however I get the case.

**Review by loginov-rocks — APPROVED** [review]

File: src/components/MovieCard/MovieCard.tsx (hunk 2 above)
Comment (psergeev) [reply src/components/MovieCard/MovieCard.tsx:12]: This is recommended way to type classes prop.

File: src/components/MovieCard/MovieCard.tsx (hunk 1 above)
Comment (psergeev) [reply src/components/MovieCard/MovieCard.tsx]: done

File: src/components/MovieCard/MovieCard.tsx (hunk 2 above)
Comment (psergeev) [reply src/components/MovieCard/MovieCard.tsx:12]: I don't like this way, but it is recommended way to type classes prop.

Comment (loginov-rocks) [reply src/components/MovieCard/MovieCard.tsx:12]: Yep, I got the case, I'm okay with that.

**Comment by loginov-rocks** [comment]
Please proceed with merge. Thank you for a great work!

**Pull request merged**